File: VocaDbModel/DataContracts/Albums/CreateAlbumContract.cs
#nullable disable

using VocaDb.Model.DataContracts.Artists;
using VocaDb.Model.Domain.Albums;

namespace VocaDb.Model.DataContracts.Albums
{
	public class CreateAlbumContract
	{
		public ArtistContract[] Artists { get; init; }

		public DiscType DiscType { get; init; }

		public LocalizedStringContract[] Names { get; init; }
	}
}

File: VocaDbWeb/Scripts/DataContracts/Album/CreateAlbumContract.ts
import { ArtistContract } from '@/DataContracts/Artist/ArtistContract';
import { LocalizedStringContract } from '@/DataContracts/Globalization/LocalizedStringContract';
import { AlbumType } from '@/Models/Albums/AlbumType';

// Corresponds to the CreateAlbumForApiContract record class in C#.
export interface CreateAlbumContract {
	artists: ArtistContract[];
	discType: AlbumType;
	names: LocalizedStringContract[];
}
